test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePollerTest.java

```diff
@@ -227,4 +230,37 @@ public void shouldFailIfASingleValueOfMultipleValuesIsNotFound() throws Exceptio
         poller.pollUntilJsonObjectFoundWithValues(URL, MEDIA_TYPE, matchValues);
     }
 
+    @Test
+    public void shouldPassHeadersToRestClientIfItsProvided() {
+
+        HttpResponsePoller testObj = new HttpResponsePoller(restClient);
+
+        final String responseText = "Condition Met";
+
+        MultivaluedMap<String, Object> arbitraryHerader = new MultivaluedMapImpl<>();
+        when(restClient.query(URL, MEDIA_TYPE, arbitraryHerader)).thenReturn(response);
+        when(response.getStatus()).thenReturn(OK.getStatusCode());
+        when(response.readEntity(String.class)).thenReturn(responseText);
+
+        final String result = testObj.withHeaders(arbitraryHerader).pollUntilFoundWithCondition(URL, MEDIA_TYPE, responseText::equals);
+
+        assertThat(result, is(responseText));
+    }
+
+    @Test
+    public void shouldIgnoreHeadersIfItsNotProvided() {
+
+        HttpResponsePoller testObj = new HttpResponsePoller(restClient);
```

Comment:
I have added headers in declarative way.... wants to change in a way having less side effects.. i can make it final then i have force headers through constructor... which i don't want... it's up to user  they wants to provide headers or not... please let me know if you still wants to make it final...